syntax = "proto3";

package user;

service UserService {
  rpc CreateUser (CreateUserRequest) returns (User) {}
  rpc GetAllUsers (Empty) returns (GetAllUsersResponse) {}
  rpc GetUserById (GetUserByIdRequest) returns (User) {}
  rpc UpdateUser (UpdateUserRequest) returns (User) {}
  rpc DeleteUser (DeleteUserRequest) returns (User) {}
}

message Empty {}

message User {
  string id = 1;
  string name = 2;
  string email = 3;
  string password = 4;
  string createdAt = 5;
  string updatedAt = 6;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
  string password = 3;
}

message GetUserByIdRequest {
  string id = 1;
}

message GetAllUsersResponse {
  repeated User users = 1;
}

message UpdateUserRequest {
  string id = 1;
  string name = 2;
  string email = 3;
  string password = 4;
}

message DeleteUserRequest {
  string id = 1;
} 